test/matrix_spec_coverage.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import re
+import yaml
+from responses import RequestsMock
+
+INTERPOLATIONS = [
+    ("%CLIENT_MAJOR_VERSION%", "r0")
+]
+
+def interpolate_str(s):
+    for interpolation in INTERPOLATIONS:
+        s = s.replace(interpolation[0], interpolation[1])
+    return s
+
+def endpoint_to_regex(s):
+    # TODO sub by with more specific REGEXes per type
+    # e.g. roomId, eventId, userId
+    return re.sub('\{[a-zA-Z]+\}', '[a-zA-Z!\.:-@#]+', s)
```

Comment:
This is bad, requests, or some other library should provide a way to directly assert whether a request matches e.g. `/room_id/{roomId}` I haven't really looked into it, but regexing is not the preferred way for sure.